src/applets/budgie-menu/backend/Tracker.vala

```diff
@@ -0,0 +1,225 @@
+			included_categories = { "Education" },
+			excluded_categories = { "Science" }
+		});
+
+		this.categories.add(new Category(_("Games")) {
+			included_categories = { "Game" }
+		});
+
+		this.categories.add(new Category(_("Graphics")) {
+			included_categories = { "Graphics" },
+			excluded_applications = {
+				"eog.desktop",
+				"org.gnome.Evince.desktop",
+				"simple-scan.desktop"
+			}
+		});
+
+		this.categories.add(new Category(_("Internet")) {
+			included_categories = { "Network" },
+			excluded_applications = { "vinagre.desktop" }
+		});
+
+		this.categories.add(new Category(_("Office")) {
+			included_categories = { "Office" },
+			excluded_applications = { "org.gnome.Dictionary.desktop", "org.gnome.Evince.desktop" }
+		});
+
+		this.categories.add(new Category(_("Programming")) {
+			included_categories = { "Development" }
+		});
+
+		this.categories.add(new Category(_("Science")) {
+			included_categories = { "Science", "Education" },
+			// LibreOffice Math is an office application, not a science application
+			excluded_applications = { "libreoffice-math.desktop" }
+		});
+
+		this.categories.add(new Category(_("Sound & Video")) {
+			included_categories = { "AudioVideo" }
+		});
+
+		this.categories.add(new Category(_("System Tools")) {
+			included_categories = { "Administration", "Settings", "System" },
+			excluded_categories = { "Games", "X-GNOME-Utilities" },
+			excluded_applications = {
+				"onboard.desktop",
+				"onboard-settings.desktop",
+				"org.gnome.baobab.desktop",
+				"org.gnome.Usage.desktop"
+			}
+		});
+
+		this.categories.add(new Category(_("Universal Access")) {
+			included_categories = { "Accessibility" }
+		});
+
+		this.categories.add(new Category(_("Utilities")) {
+			included_categories = { "X-GNOME-Utilities" }
+		});
+
+		// Create our misc category, but don't add it to the list until it actually has apps in it
+		var misc_category = new Category(_("Other"), true) {
+			excluded_categories = { "Core", "Screensaver", "Settings" },
+			// All of these should be in Utilities
+			excluded_applications = { "htop.desktop", "onboard.desktop", "org.gnome.FileRoller.desktop", "org.gnome.font-viewer.desktop" }
+		};
+
+		// Iterate over all registered AppInfos and try to put them in categories
+		foreach (var app in AppInfo.get_all()) {
+			unowned var desktop_app = app as DesktopAppInfo;
+			if (desktop_app == null) {
+				continue;
+			}
+
+			// Check if this is a control center panel
+			var control_center = "budgie-control-center";
+			bool is_control_center_panel = (
+				app.get_commandline() != null &&
+				control_center in app.get_commandline() &&
+				app.get_commandline().length != control_center.length
+			);
+
+			// Check if we should not add this application to the store
+			bool should_skip = (!desktop_app.should_show() && !is_control_center_panel) || // CC items are handled by the UI
+								(desktop_app.get_boolean("Terminal"));
+
+			if (should_skip) {
+				continue;
+			}
```

Comment:
I can't say I am a fan of this logic being here. The idea of the tracker was solely to provide a central store for appinfo, with unified search logic. If something should be excluded on the presentation layer, like Budgie Menu, it should be done at the presentation layer.

Reply:
The problem is that the .desktop files for BCC panels currently have `NoDisplay=true` or whatever the syntax is. Should that then be changed and this check here removed?

Reply:
Think I was misunderstanding the purpose of this. Feel free to ignore.